ui/preference/list_preference.go

```diff
@@ -0,0 +1,202 @@
+	}
+}
+
+func (lp *ListPreference) Title(titleStrKey string) *ListPreference {
+	lp.titleStrKey = titleStrKey
+	return lp
+}
+
+func (lp *ListPreference) PostiveButton(strkey string, clicked func()) *ListPreference {
+	lp.positiveButtonStrKey = strkey
+	lp.positiveButtonClicked = clicked
+
+	return lp
+}
+
+func (lp *ListPreference) NegativeButton(strkey string, clicked func()) *ListPreference {
+	lp.negativeButtonStrKey = strkey
+	lp.negativeButtonClicked = clicked
+
+	return lp
+}
+
+func (lp *ListPreference) Clickable() *widget.Clickable {
+	return lp.clickable
+}
+
+func (lp *ListPreference) Handle() {
+
+	for lp.clickable.Clicked() {
+		initialValue := lp.wallet.ReadStringConfigValueForKey(lp.preferenceKey)
+		if initialValue == "" {
+			initialValue = lp.defaultValue
+		}
+
+		lp.initialValue = initialValue
+		lp.currentValue = initialValue
+		lp.IsShowing = true
+	}
+
+	for lp.negativeButton.Button.Clicked() {
+		lp.setValue(lp.initialValue) // reset value
+		lp.IsShowing = false
+
+		lp.negativeButtonClicked()
+	}
+
+	for lp.positiveButton.Button.Clicked() {
+		lp.setValue(lp.optionsRadioGroup.Value) // set value
+		lp.IsShowing = false
+
+		lp.positiveButtonClicked()
+	}
+
+	for lp.optionsRadioGroup.Changed() {
+		lp.currentValue = lp.optionsRadioGroup.Value
+	}
+}
+
+func (lp *ListPreference) setValue(value string) {
+	lp.wallet.SaveConfigValueForKey(lp.preferenceKey, value)
+	values.SetUserLanguage(value)
+}
+
+func (lp *ListPreference) Layout(gtx layout.Context, body layout.Dimensions) layout.Dimensions {
+	return layout.Stack{}.Layout(gtx,
+		layout.Expanded(func(gtx layout.Context) layout.Dimensions {
+			return body
+		}),
+		layout.Stacked(func(gtx layout.Context) layout.Dimensions {
+			return lp.modal(gtx)
+		}),
+	)
+}
+
+func (lp *ListPreference) modal(gtx layout.Context) layout.Dimensions {
+	w := []layout.Widget{
+		func(gtx layout.Context) layout.Dimensions {
+			txt := lp.theme.H6(values.String(lp.titleStrKey))
+			txt.Color = lp.theme.Color.Text
+			return txt.Layout(gtx)
+		},
+		func(gtx layout.Context) layout.Dimensions {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx, lp.layoutItems()...)
+		},
+		func(gtx layout.Context) layout.Dimensions {
+			return layout.E.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx, lp.layoutButtons()...)
+			})
+		},
+	}
+
+	lp.optionsRadioGroup.Value = lp.currentValue
+
+	return lp.theme.Modal().Layout(gtx, w, 1050)
+}
+
+func (lp *ListPreference) layoutItems() []layout.FlexChild {
+
+	// sort keys to keep order when refreshed
+	keys := make([]string, 0)
+	for k := range lp.items {
+		keys = append(keys, k)
+	}
+
+	sort.Slice(keys, func(i int, j int) bool { return keys[i] < keys[j] })
+
+	items := make([]layout.FlexChild, 0)
+	for _, k := range keys {
+		radioItem := layout.Rigid(lp.theme.RadioButton(lp.optionsRadioGroup, k, values.String(lp.items[k])).Layout)
```

Comment:
This seems to be an implementation to selection options using radio buttons. Maybe RadioSelection would be a more descriptive name for the widget.

Reply:
It's not a widget, it's specifically for list preferences and it saves the select value to the config database.